libs/blocks/global-navigation/utilities/utilities.js

```diff
@@ -243,3 +288,46 @@ export function processMartechAttributeMetadata(html) {
   });
   return null;
 }
+
+export async function fetchAndProcess({ url, message, shouldDecorateLinks = true } = {}) {
```

Comment:
Could we make the method name a bit more specific? It's not immediately clear what is fetched and processed, could be images, some data, fragments, etc.

Also, I'd make the `message` argument name more specific, it's an error message.

Reply:
What do you think about: `fetchAndProcessPlainHtml` ?
This should be useful for any place we fetch anything `.plain.html` and enrich it with all the neccesary things (localization, placeholders, federated pictures etc...)

Reply:
Sounds reasonable, sure!